src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -222,5 +234,13 @@ private static JObject ReadJson(string packageSpecPath)
 
             return json;
         }
+
+        private static HashSet<RestoreOutputType> RestorableTypes = new HashSet<RestoreOutputType>()
```

Comment:
The restorable decision here is related to how NuGet will handle these projects, not how the projects want to be handled.